lib/grape/dsl/inside_route.rb

```diff
@@ -91,10 +91,9 @@ def handle_passed_param(params_nested_path, has_passed_children = false, &_block
           route_options_params = options[:route_options][:params] || {}
           type = route_options_params.dig(key, :type)
           has_children = route_options_params.keys.any? { |k| k != key && k.start_with?(key) }
-
           if type == 'Hash' && !has_children
             {}
-          elsif type == 'Array' || type&.start_with?('[')
+          elsif type == 'Array' || type && type.start_with?('[') && !type.include?(',')
```

Comment:
You are right @dblock that working with Strings here is not ideal. I don't know enough to know if there is a better alternative.